pkg/agent/plugin/workloadattestor/systemd/systemd_posix.go

```diff
@@ -105,13 +84,23 @@ func getSystemdUnitInfo(ctx context.Context, pid uint) (unitInfo, error) {
 	}
 
 	obj := conn.Object(systemdDBusInterface, unitPath)
-	return DBusUnitObject{obj}, nil
+
+	id, err := getStringProperty(obj, "Id")
+	if err != nil {
+		return nil, err
+	}
+	fragmentPath, err := getStringProperty(obj, "FragmentPath")
+	if err != nil {
+		return nil, err
+	}
+
+	return &DBusUnitInfo{id, fragmentPath}, nil
 }
 
-func getStringProperty(obj DBusUnitObject, prop string) (string, error) {
+func getStringProperty(obj dbus.BusObject, prop string) (string, error) {
 	propVariant, err := obj.GetProperty(systemdDBusInterface + ".Unit." + prop)
 	if err != nil {
-		return "", err
+		return "", status.Errorf(codes.Internal, "Error getting value for %v: %v", prop, err)
```

Comment:
Status error messages are by convention lowercase. Also, using specific format directives when available gives us better diagnostics with `go vet` so `%s` is preferred for strings.
```suggestion
		return "", status.Errorf(codes.Internal, "error getting value for %s: %v", prop, err)
```